packages/automated_actions_utils/automated_actions_utils/openshift_client.py

```diff
@@ -246,13 +254,41 @@ def run_job(
         wait_for_completion: bool = True,
     ) -> None:
         """Run a Kubernetes Job and optionally wait for its completion."""
-        job_api = self.dyn_client.resources.get(api_version="batch/v1", kind="Job")
         log.info(f"Creating Job '{job.metadata.name}' in namespace '{namespace}'")
-        job_api.create(
-            body=self.k8s_api_client.sanitize_for_serialization(job),
+        self.batch_v1.create_namespaced_job(
             namespace=namespace,
+            body=self.k8s_api_client.sanitize_for_serialization(job),
         )
         if wait_for_completion:
             self.job_wait(
                 job_name=job.metadata.name, namespace=namespace, timeout_seconds=10 * 60
             )
+
+    def trigger_cronjob(
+        self, namespace: str, cronjob: str, annotations: dict[str, str] | None = None
+    ) -> None:
+        try:
+            k8s_cronjob: V1CronJob = self.batch_v1.read_namespaced_cron_job(
+                name=cronjob, namespace=namespace
+            )
+        except ApiException as err:
+            if err.status == http.HTTPStatus.NOT_FOUND:
+                raise OpenshiftClientResourceNotFoundError(
+                    f"CronJob {cronjob} does not exist in namespace {namespace}"
+                ) from err
+            raise
+
+        # make mypy happy
+        assert k8s_cronjob.spec is not None
+        assert k8s_cronjob.spec.job_template is not None
+
+        job = V1Job(
+            api_version="batch/v1",
+            kind="Job",
+            metadata=V1ObjectMeta(
+                name=f"aa-triggered-{cronjob[:41]}-{uuid.uuid4().hex[:8]}",
```

Comment:
now we have second use case of add random suffix to job name and ensure job name length within 63, https://github.com/app-sre/automated-actions/blob/de07323d71604c35097139f182c4e773b1fbe51f/packages/automated_actions_utils/automated_actions_utils/openshift_client.py#L115

we should extract a common function given a name, return a valid job name with random suffix, so no need to have magic numbers.

doc: https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/

* max cronjob name length is 52
* max job name length is 63

similar to https://github.com/app-sre/qontract-reconcile/pull/5258#discussion_r2464391956

Reply:
Here we go https://github.com/app-sre/sretoolbox/pull/147

Reply:
Done in ea8f6a3